deliver/lib/deliver/upload_app_clip_default_experience_metadata.rb

```diff
@@ -102,5 +120,49 @@ def upload_subtitle_localizations(app_clip_default_experience:, subtitle_localiz
       app_info_worker.batch_enqueue(localized_subtitle_attributes_by_locale.keys)
       app_info_worker.start
     end
+
+    # Handle the case where an app clip exists, but the user did not specify any app clip default
+    # experience metadata options. We check if the current editable version already has the app clip
+    # default experience metadata. If not, copy over the live version's app clip default
+    # experience metadata. This mimics the default behavior of creating an App Store version from
+    # the ASC UI.
+    #
+    # This function will also produce warnings, but not outright fail, when it finds missing app
+    # clip metadata.
+    #
+    # As of 2022-05-18, App Store versions created by the ASC API do not "carry over" the live
+    # version's app clip default experience metadata, so we handle this for the user by default.
+    #
+    def copy_live_version_app_clip_default_experience_metadata(app:, platform:, edit_version:, app_clip:)
+      edit_version_default_experience = edit_version.app_clip_default_experience
+      if !edit_version_default_experience
+        live_version = fetch_live_app_info(app, platform)
+        # no live version to carry over metadata from
+        return if live_version.nil?
+
+        live_version_default_experience = live_version.app_clip_default_experience
+        # no live version default experience to carry over metadata from
+        return if live_version_default_experience.nil?
+
+        # create a default experience and use the live version default experience as a "template"
+        Spaceship::ConnectAPI::AppClipDefaultExperience.create(app_clip_id: app_clip.id, app_store_version_id: edit_version.id, template_default_experience_id: live_version_default_experience.id)
+
+        # TODO: if the above code using the "template" doesn't work, we can use the following to manually copy the values
+        #
+        # default_experience.update(attributes: { action: live_version_default_experience.action })
+        # UI.message("Added app clip default experience action from live version #{live_version.version}")
+
+        # # copy the live version localizations
+        # live_version_localizations = Spaceship::ConnectAPI::AppClipDefaultExperience.get(app_clip_default_experience_id: live_version_default_experience.id, includes: 'appClipDefaultExperienceLocalizations').app_clip_default_experience_localizations
+
+        # live_version_localizations.each do |live_localization|
+        #   attributes = { subtitle: live_localization.subtitle, locale: live_localization.locale }
+        #   Spaceship::ConnectAPI::AppClipDefaultExperienceLocalizations.create(default_experience_id: default_experience.id, attributes: attributes)
+        # end
```

Comment:
I'm planning to remove this commented code once I can confirm the default experience template option works as I expect.